### PR TITLE
Fy/dev

### PR DESCRIPTION
**Pull request by 2020dfff**
### Summary :memo:
MAS repo analysis and part of the visualize

### Details
_Describe more what you did on changes._
1. Write the MAS topology generate logic for 5 types of topology;
2. Visualize and basic analysis;

### Bugfixes :bug: (delete if dind't have any)
-

### Checks
-

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -353,3 +353,7 @@ poetry.toml
 pyrightconfig.json
 
 # End of https://www.toptal.com/developers/gitignore/api/python,jupyternotebooks
+
+# self-defined
+plot/
```
Comment (jjkysy) [inline .gitignore]: 不要改这个文件。

Comment (2020dfff) [reply .gitignore]: Rollback already ✅

**Comment by 2020dfff** [comment]
Rollback .gitignore file already ✅

**Review by jjkysy — APPROVED** [review]
LGTM

**Comment by jjkysy** [comment]
LGTM

**Pull request merged**